### DISCUSSION TITLE
PooledHashSet(Of T) with different equality check.

### DISCUSSION THREAD

**Discussion by AdamSpeight2008** [dotnet/roslyn#49570] [Q&A] [answered]
Is there a way to specify how a `PooledHashSet(Of T)` does it equality check?
I would to use `ReferenceEqualityComparer.Instance`

**Comment by CyrusNajmabadi** [comment, marked as answer]
Yes.  Call `public static ObjectPool<PooledHashSet<T>> CreatePool(IEqualityComparer<T> equalityComparer)` to create a pool wit that comparer.  Then retrieve/return instances to that pool.
